File: VB/CalculatedAlias.Module/Product.vb
Imports System.ComponentModel

Imports DevExpress.Xpo
Imports DevExpress.Data.Filtering

Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Xpo
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.Persistent.Validation


Namespace CalculatedAlias.Module
    <CalculatedPersistentAlias("ProductName", "ProductNameAlias"), DefaultClassOptions> _
    Public Class Product
        Inherits BaseObject

        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub

        Private Shared productNameFormat As String = "{Manufacturer} {Model}"
        Private Shared _productNameAlias As String = "concat(Manufacturer, Model)"

        Public ReadOnly Property ProductName() As String
            Get
                Return ObjectFormatter.Format(productNameFormat, Me, EmptyEntriesMode.RemoveDelimeterWhenEntryIsEmpty)
            End Get
        End Property
        <Browsable(False)> _
        Public Shared ReadOnly Property ProductNameAlias() As String
            Get
                Return _productNameAlias
            End Get
        End Property

        Public Shared Sub SetProductNameFormat(ByVal productNameFormat As String, ByVal productNameAlias As String)
            If Not String.IsNullOrEmpty(productNameFormat) Then
                Product.productNameFormat = productNameFormat
            End If
            If Not String.IsNullOrEmpty(productNameAlias) Then
                Product._productNameAlias = productNameAlias
            End If
        End Sub

        Public Property Manufacturer() As String
            Get
                Return GetPropertyValue(Of String)("Manufacturer")
            End Get
            Set(ByVal value As String)
                SetPropertyValue(Of String)("Manufacturer", value)
            End Set
        End Property
        Public Property Model() As String
            Get
                Return GetPropertyValue(Of String)("Model")
            End Get
            Set(ByVal value As String)
                SetPropertyValue(Of String)("Model", value)
            End Set
        End Property
        Public Property Revision() As String
            Get
                Return GetPropertyValue(Of String)("Revision")
            End Get
            Set(ByVal value As String)
                SetPropertyValue(Of String)("Revision", value)
            End Set
        End Property
    End Class

End Namespace

File: CS/CalculatedAlias.Module/Product.cs
using System;
using System.ComponentModel;

using DevExpress.Xpo;
using DevExpress.Data.Filtering;

using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Xpo;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Persistent.Validation;


namespace CalculatedAlias.Module {
    [CalculatedPersistentAlias("ProductName", "ProductNameAlias"), DefaultClassOptions]
    public class Product : BaseObject {
        public Product(Session session) : base(session) { }

        private static string productNameFormat = "{Manufacturer} {Model}";
        private static string productNameAlias = "concat(Manufacturer, Model)";
        
        public string ProductName {
            get { return ObjectFormatter.Format(productNameFormat, this, EmptyEntriesMode.RemoveDelimiterWhenEntryIsEmpty); }
        }
        [Browsable(false)]
        public static string ProductNameAlias {
            get { return productNameAlias; }            
        }

        public static void SetProductNameFormat(string productNameFormat, string productNameAlias) {
            if(!string.IsNullOrEmpty(productNameFormat)) {
                Product.productNameFormat = productNameFormat;
            }
            if(!string.IsNullOrEmpty(productNameAlias)) {
                Product.productNameAlias = productNameAlias;
            }
        }

        public string Manufacturer {
            get { return GetPropertyValue<string>("Manufacturer"); }
            set { SetPropertyValue<string>("Manufacturer", value); }
        }
        public string Model {
            get { return GetPropertyValue<string>("Model"); }
            set { SetPropertyValue<string>("Model", value); }
        }
        public string Revision {
            get { return GetPropertyValue<string>("Revision"); }
            set { SetPropertyValue<string>("Revision", value); }
        }
    }

}
